include/lbann/data_store/jag_io.hpp

```diff
@@ -0,0 +1,193 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2016, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef __JAG_IO_HPP__
+#define __JAG_IO_HPP__
+
+#include "lbann_config.hpp"
+
+#ifdef LBANN_HAS_CONDUIT
+#include "conduit/conduit.hpp"
+#include "conduit/conduit_relay.hpp"
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace lbann {
+
+class jag_io {
+ public:
+
+ /**
+  * NOTE: some methods below take variables "string node_name" while others
+  *       take "string key." My convention is that "node_name" indicated
+  *       a fully qualified name, i.e, it begins with the sample id:
+  *          0/field_1/field_2
+  *       on the other hand, "key" does not contain the sample id:
+  *          field_1/field_2
+  */
+
+/// WARNING! CAUTION! BE ADVISED!
+/// cut-n-paste from data_reader_jag_conduit; this is
+/// fragile -- probably best to but these in a small
+/// file that both reader and store can then include
+/// @todo
+using ch_t = double; ///< jag output image channel type
+using scalar_t = double; ///< jag scalar output type
+using input_t = double; ///< jag input parameter type
+
+  //! ctor
+  jag_io();
+
+  //! copy ctor
+  jag_io(const jag_io&) = default;
+
+  //! operator=
+  jag_io& operator=(const jag_io&) = default;
+
+  //@todo: this causes a compile error ... why?
+  //jag_io * copy() const { return new jag_io(*this); }
```

Comment:
I do not have any error with clang/4.0.0 when I uncomment this line.

Reply:
I got the error with gnu; from the error message it appears to be related to the conduit headers. I just tried compiling with clang on catalyst, but the build failed early on. I've sent Tom the error messages. I've started work on data_store_jag_conduit, and ran into the same compile error with copy().

Reply:
By any chance, are you using the conduit not built for the target platform?